### DISCUSSION TITLE
jose.jwtVerify - ReferenceError: CryptoKey is not defined

### DISCUSSION THREAD

**Discussion by samuel-casasdray** [panva/jose#411] [Q&A]
On local (ng serve) it works perfectly with this code
jose.jwtVerify(observer['token'], new TextEncoder().encode(environment.jwtKey)).then(...)

But when I build the application to put it on the server, I got an error :
`ReferenceError: CryptoKey is not defined`

So I don't really know how to solve that...
I even do a npm i crypto just in case, but it doesn't solve the problem

Version:
angular : 13.2.1
jose : 4.8.1

**Comment by panva** [comment]
Prepare a working example that can be built and run if you wish to get assistance please.

**Reply by samuel-casasdray** [reply]
How can i do that ?

**Reply by panva** [reply]
Maybe create a reproduction repository with the code and the steps needed to be taken described in it.

**Reply by samuel-casasdray** [reply]
ok i will try

**Comment by pickledbrill** [comment]
Have the same issue with CryptoKey.

**Reply by pickledbrill** [reply]
Find [this article](https://developer.mozilla.org/en-US/docs/Web/API/CryptoKey) describes that CryptoKey only works in HTTPS. Adding SSL certificate just solved the issue.

**Comment by Lukunlin** [comment]
I met the same problem as you, but in fact he is not a problem。

because this API can only run in localhost and HTTPS environments.

therefore, the scenario where you make an error may be running in the HTTP environment

**Reply by panva** [reply]
https://developer.mozilla.org/en-US/docs/Web/API/Web_Crypto_API

<img width="784" alt="image" src="https://user-images.githubusercontent.com/241506/189907432-84a2bdfa-d7fa-4a6f-8385-fc5f371ea3fa.png">


https://developer.mozilla.org/en-US/docs/Web/Security/Secure_Contexts

**Comment by yerco** [comment]
I tried
```ts
import { Crypto } from '@peculiar/webcrypto'
global.crypto = new Crypto()
```
but for what i was doing a better alternative was
```ts
const cryptoAlt = require('crypto').webcrypto
const cryptoKey = require('crypto').webcrypto.CryptoKey
global.crypto = cryptoAlt
global.CryptoKey = cryptoKey
```
